### PR TITLE
#49 uncompressed dependencies

### PR DESCRIPTION
**Pull request by klausbayrhammer**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -37,6 +37,11 @@ var main = function () {
     abbr: 'c',
     help: 'directory where dependencies will be cached'
   });
+  parser.option('deflate', {
```
Comment (JoshSchreuder) [inline index.js]: Is there any way to get this working without providing a value?

EG. my command line is:
`npm-cache install --cacheDirectory c:\npm-cache --deflate npm --production`

But running that gives error:
```
'--deflate' expects a value


Usage: C:\Program Files\nodejs\node.exe index.js install [options]

Options:
   -c, --cacheDirectory   directory where dependencies will be cached  [D:\Users\dev_user\.package_cache]
   -v, --version          displays version info and exit
   -r, --forceRefresh     force installing dependencies from package manager without cache  [false]
```

So I have to run it as
`npm-cache install --cacheDirectory c:\npm-cache --deflate y npm --production`

Which seems a little odd.

Also the help file text doesn't list the new flag :smile:

File: index.js [hunk 2]
```diff
@@ -60,6 +65,7 @@ var main = function () {
     '\tnpm-cache install bower --allow-root composer --dry-run\t# install bower with allow-root, and composer with --dry-run',
     '\tnpm-cache install --cacheDirectory /home/cache/ bower \t# install components using /home/cache as cache directory',
     '\tnpm-cache install --forceRefresh  bower\t# force installing dependencies from package manager without cache',
+    '\tnpm-cache install --deflate npm\t# do not gzip the dependencies when storing them',
```
Comment (JoshSchreuder) [inline index.js]: This doesn't work, as per above comment

File: cacheDependencyManagers/cacheDependencyManager.js [hunk 3]
```diff
@@ -72,40 +70,50 @@ CacheDependencyManager.prototype.archiveDependencies = function (cacheDirectory,
...
-    .on('error', onError)
-    .pipe(extractor);
+    fs.createReadStream(cachePath)
+        .on('error', onError)
+        .pipe(extractor);
+  } else {
+    fstream.Reader(cachePath)
+        .on('error', onError)
+        .on('end', onEnd)
+        .pipe(fstream.Writer(targetPath));
```
Comment (JoshSchreuder) [inline cacheDependencyManagers/cacheDependencyManager.js]: Should this be `process.cwd()` instead of `targetPath`?

**Comment by JoshSchreuder** [comment]
:+1: Works well, just had to change the commented line as `targetPath` wasn't defined.

**Comment by swarajban** [comment]
Hey sorry for the delay, will review early this week

**Comment by JonWallsten** [comment]
I just did a quick test with the this PR.

Without flag:
Install: 05.03.1
Install from cache(With clean): 02.28.6
Total: 07.31.7

With --noArchive:
Install: 06.48.8
Install from cache(With clean): 02.09.4
Total: 8.58.2

Would be cool to compare --noArchive with using tar-fs instead of node-tar.
Another thing would be to lower the compression level no none, if it's possible.

**Comment by JoshSchreuder** [comment]
So I ran a few trials (Windows 8, Node v5.7.0):

| | Attempt 1 | Attempt 2 | Attempt 3 | Average |
| ---------------- | ------------- | ------------- | ------------- | ------------- |
| **0.6.0 Clean** | 45.02s  | 40.62s  | 47.49s | 44.37s |
| **0.6.0 Cached**  | 26.20s  | 24.69s | 25.62s | 25.50s |
| | | | 
| **no-archive Clean**  | 43.20s  | 41.43s | 49.37s | 44.66s |
| **no-archive Cached**  | 17.57s  | 17.83s | 15.66s | 17.02s |
| | | |
| **tar-fs Clean**  | 39.01s  | 43.09s | 42.65s | 41.58s |
| **tar-fs Cached**  | 16.82s  | 15.73s | 15.17s | 15.90s |

The one thing I didn't cater for was network latency, so the clean runs may have some margin of error to them. You could eliminate this by running a network cache, but I couldn't be bothered :smile: I expect the cached runs are accurate.

It looks like there is significant improvement to be made by eliminating or improving unarchiving. `tar-fs` performed better than the stock `node-tar` in this regard.

**Comment by swarajban** [comment]
Hi, can you re-create this PR against the latest `feature/pull-requests` branch. Looks like I messed something up when merging in the last PR, so I just re-created this branch.

**Pull request closed without merging**